src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -1267,21 +1290,50 @@ static void icap_clean_bitstream_axlf(struct platform_device *pdev)
 	icap_clean_axlf_section(icap, PARTITION_METADATA);
 }
 
-static uint32_t convert_mem_type(const char *name)
+enum MEM_TAG convert_mem_tag(const char *name)
 {
 	/* Don't trust m_type in xclbin, convert name to m_type instead.
-	 * m_tag[i] = "HBM[0]" -> m_type = MEM_HBM
-	 * m_tag[i] = "DDR[1]" -> m_type = MEM_DRAM
-	 *
-	 * Use MEM_DDR3 as a invalid memory type. */
-	enum MEM_TYPE mem_type = MEM_DDR3;
+	 * m_tag[i] = "HBM[0]" -> m_type = MEM_TAG_HBM
+	 * m_tag[i] = "DDR[1]" -> m_type = MEM_TAG_DRAM
+	 */
+	enum MEM_TAG mem_tag = MEM_TAG_INVALID;
 
 	if (!strncasecmp(name, "DDR", 3))
-		mem_type = MEM_DRAM;
+		mem_tag = MEM_TAG_DDR;
+	else if (!strncasecmp(name, "PLRAM", 5))
+		mem_tag = MEM_TAG_PLRAM;
 	else if (!strncasecmp(name, "HBM", 3))
-		mem_type = MEM_HBM;
+		mem_tag = MEM_TAG_HBM;
 	else if (!strncasecmp(name, "bank", 4))
-		mem_type = MEM_DRAM;
+		mem_tag = MEM_TAG_DDR;
+	else if (!strncasecmp(name, "HOST[0]", 7))
+		mem_tag = MEM_TAG_HOST;
+
+	return mem_tag;
+}
+
+static enum MEM_TYPE convert_mem_type(const char *name)
```

Comment:
I am not sure this is a good idea, it is very convenient to group the memory tags by their types and perform actions on them as a group rather than spelling out each individual name. For example one of the remaining uses of the convert_mem_type function perform actions depends on the ECC memory type (icap.c::icap_save_calib). If we remove this function we would need to list out every memory tag that is a DRAM type. We would end up with a macro to group the types together and at that point we might as well keep the function around.

Reply:
m_type is not accurate and confusing. We should use tag string to category/group banks. you already created a tag enum to which can category/group tag strings, correct? Why would we still need to group by m_type? 
For icap_save_calib, isn't MEM_TAG_DDR good enough? convert_mem_tag() returns MEM_TAG_DDR for both 'bank' and 'DDR', correct?

Reply:
Okay, I though once we converted the memory types they were valid. I'll refactor the functions to use the tag instead.